src/main/java/address/storage/StorageManager.java

```diff
@@ -136,11 +136,49 @@ public void saveDataToFile(File file, ReadOnlyAddressBook data){
     }
 
     /**
+     * Raises FileSavingExceptionEvent
+     */
+    @Subscribe
+    public void handleSavePrefsRequestEvent(SavePrefsRequestEvent spre) {
+        logger.info("Save prefs request received: {}", spre.prefs);
+        savePrefsToFile(spre.file, spre.prefs);
+    }
+
+    /**
+     * Raises FileSavingExceptionEvent if there was an error during saving or data conversion.
+     */
+    public void savePrefsToFile(File file, UserPrefs prefs) {
+        try {
+            FileUtil.writeToFile(file, JsonUtil.toJsonString(prefs));
+        } catch (IOException e) {
+            raise(new FileSavingExceptionEvent(e, file));
+        }
+    }
+
+    public static UserPrefs loadPrefsFromFile(File prefsFile) {
+        UserPrefs prefs = new UserPrefs();
+
+        if (!FileUtil.isFileExists(prefsFile)) {
+            return prefs;
+        }
+
+        try {
+            logger.debug("Attempting to load prefs from file: " + prefsFile);
+            prefs = JsonUtil.fromJsonString(FileUtil.readFromFile(prefsFile), UserPrefs.class);
+        } catch (IOException e) {
+            logger.debug("Error loading prefs from file: {}", e);
+        }
+
+        return prefs;
+    }
+
+    /**
      * Loads the data from the local data file (based on user preferences).
      */
     public void start() {
         logger.info("Starting storage manager.");
-        loadDataFromFile(userPrefs.getSaveLocation());
+        loadDataFromFile(modelManager.getPrefs().getSaveLocation());
```

Comment:
It's better for StorageManger to keep a copy of preferences rather than get it from model manager. Storage location is StorageManager's business, not ModelManager's.